cf-browser/src/main/java/org/eclipse/californium/tools/GUIController.java

```diff
@@ -801,14 +803,22 @@ private void showResponse(Response response) {
 				image = new Image(imgIS);
 			}
 			mediaTypeView.setImage(image);
-			String text = response.getPayloadString();
-			if (format == MediaTypeRegistry.APPLICATION_LINK_FORMAT) {
-				Set<WebLink> webLinks = LinkFormat.parse(text);
-				StringBuilder links = new StringBuilder();
-				for (WebLink link : webLinks) {
-					links.append(link).append(StringUtil.lineSeparator());
+			String text = "";
+			if (format == MediaTypeRegistry.APPLICATION_OCTET_STREAM) {
+				text = StringUtil.byteArray2Hex(payload);
+			} else if (format == MediaTypeRegistry.APPLICATION_CBOR) {
+				CBORObject element = CBORObject.DecodeFromBytes(payload);
+				text = element.toString();
+			} else {
```

Comment:
shouldn't the fallback better be to handle the payload as opaque binary data (which should work in any case)?

Reply:
Not sure, if I understand, what you mean?
Should we add a "binary fallback" in case of failing conversion?

Reply:
That is also a good idea, I guess.
What I meant was that the `else` branch should probably be to handle the payload as opaque bytes